pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -520,12 +520,14 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 						Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
 					})
 
-					It("should not make any changes to the CAPI machine", func() {
-						resourceVersion := capiMachine.GetResourceVersion()
-						Consistently(k.Object(capiMachine), timeout).Should(
-							HaveField("ResourceVersion", Equal(resourceVersion)),
-						)
-					})
+					// We now set finalizers regardless, so this does not work any more.
```

Comment:
Because we changed `fetchCAPIInfraResources` to return a Nil `InfraMachine` rather than error.

We changed this because we must delete the inframachine first - then the capi machine's deletion is unblocked. 

This needs more thought at a later date, but works and is tested on cluster for now.

Reply:
Won't block merge on it, but we should get a bug made soon and put a reference in the code.